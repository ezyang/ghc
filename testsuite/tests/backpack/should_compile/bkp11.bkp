package p where
    signature H where
        data S
    signature H2 where
        data T
    module A where
        import H
        import H2
        data Z = Z S T

package q where
    include p requires (H, H2 as H)
    module B where
        import H
        import A
        f :: S -> T -> Z
        f = Z
