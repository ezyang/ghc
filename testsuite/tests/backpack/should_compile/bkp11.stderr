[1 of 2] Processing p
unit p(hole:H, hole:H2)
  [1 of 3] Compiling H[sig]           ( p/H.hsig, nothing )
  [2 of 3] Compiling H2[sig]          ( p/H2.hsig, nothing )
  [3 of 3] Compiling A                ( p/A.hs, nothing )
[2 of 2] Processing q

bkp11.bkp:11:1: error:
    Duplicate dest (tell us if you need this to be supported) H
